### PR TITLE
MG-2075 - Add guest relation

### LINKED ISSUE #2075: Feature: Policy for Domain member to restrict on creation new entities such as things, channels and groups in domain
### Is your feature request related to a problem? Please describe.

Domain member could not restricted to create new entities such as things, channels and groups in a domain

### Describe the feature you are requesting, as well as the possible use case(s) for it.

Requesting a policy for Domain member to create new entities such as things, channels and groups in a domain

### Indicate the importance of this feature to you.

Must-have

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by nyagamunene**
# What type of PR is this?
This is a feature because it adds a policy to a domain member to restrict the creation of new entities such as things, channels and groups in a domain

## What does this do?
It adds a guest relation policy to a domain member to restrict the creation of new entities such as things, channels and groups in a domain

## Which issue(s) does this PR fix/relate to?

- Related Issue #2075 
- Resolves #2075 

## Have you included tests for your changes?
No

## Did you document any new/modified feature?
No

### CODE REVIEW

File: users/service.go [hunk 1]
```diff
@@ -113,6 +113,7 @@ func (svc service) IssueToken(ctx context.Context, identity, secret, domainID st
 
 	token, err := svc.auth.Issue(ctx, &magistrala.IssueReq{UserId: dbUser.ID, DomainId: &d, Type: uint32(auth.AccessKey)})
 	if err != nil {
+		fmt.Println(err.Error())
```
Comment (rodneyosodo) [inline users/service.go]: Remove debug statements

Comment (arvindh123) [inline users/service.go]: Remove this log line

**Review by arvindh123 — CHANGES_REQUESTED** [review]

**Review by arvindh123 — DISMISSED** [review]

File: auth/service.go [hunk 2]
```diff
@@ -575,6 +575,8 @@ func SwitchToPermission(relation string) string {
 		return ViewPermission
 	case MemberRelation:
 		return MembershipPermission
+	case GuestRelation:
```
Comment (rodneyosodo) [inline auth/service.go:578]: Update this test too `TestSwitchToPermission`

Comment (rodneyosodo) [reply auth/service.go:578]: also `TestCheckRelation`

**Review by rodneyosodo — CHANGES_REQUESTED** [review]
Update SDK godocs

**Review by rodneyosodo — DISMISSED** [review]

**Review by arvindh123 — CHANGES_REQUESTED** [review]
@nyagamunene , Please update SDK godocs   
like how we have added for contributor relation  https://github.com/absmach/magistrala/pull/2237/files#diff-b5c7528ff157413346fba986ef7b76e891ac8088631bdebd3fcae8b30338ddb1

**Review by rodneyosodo — DISMISSED** [review]

**Review by arvindh123 — DISMISSED** [review]
LGTM

**Review by rodneyosodo — DISMISSED** [review]

**Review by dborovcanin — APPROVED** [review]

**Pull request merged**